components/list/list-item-checkbox-mixin.js

```diff
@@ -0,0 +1,123 @@
+import { css, html } from 'lit-element/lit-element.js';
+import { checkboxStyles } from '../inputs/input-checkbox-styles.js';
+import { getUniqueId } from '../../helpers/uniqueId.js';
+
+export const ListItemCheckboxMixin = superclass => class extends superclass {
+
+	/**
+	 * https://lit-element.polymer-project.org/guide/properties#declare
+	 *
+	 * @readonly
+	 * @static
+	 */
+	static get properties() {
+		return {
+			disabled: {type: Boolean },
+			key: { type: String, reflect: true },
+			selectable: {type: Boolean },
+			selected: { type: Boolean, reflect: true }
+		};
+	}
+
+	static get styles() {
+		return [ checkboxStyles, css`
+		.list-item-checkbox {
```

Comment:
These classes will be defined in the `d2l-generic-list-item` so you just have to extend the class `control-area` and `control-action-area`